src/routes/write-exercise/WriteExerciseScreen.tsx

```diff
@@ -78,23 +70,26 @@ const WriteExerciseScreen = ({ navigation, route }: WriteExerciseScreenPropsType
         numberOfWords={docsLength}
       />
 
-      {documents && document ? (
-        <ScrollView contentContainerStyle={{ flex: 1 }} keyboardShouldPersistTaps='always'>
-          <ImageCarousel images={document.document_image} />
-          <AudioPlayer document={document} disabled={!hintsEnabled} />
-          <AnswerSection
-            currentDocumentNumber={currentDocumentNumber}
-            setCurrentDocumentNumber={setCurrentDocumentNumber}
-            documents={documents}
-            finishExercise={finishExercise}
-            tryLater={tryLater}
-            disciplineId={id}
-            setHintsEnabled={setHintsEnabled}
-          />
-        </ScrollView>
-      ) : (
-        <Spinner />
-      )}
+      <Loading isLoading={loading}>
```

Comment:
Wouldn't it be possible to make a component handling both errors and loading? Currently there is quite some duplicated code with `<ErrorMessage error={error} refresh={refresh} />`.

Reply:
done